contracts/extensions/StreamingPayments.sol

```diff
@@ -0,0 +1,445 @@
+  /// @param _adminChildSkillIndex The index linking the adminPermissionDomainId to the domainId
+  /// @param _domainId The domain out of which the streaming payment will be paid
+  /// @param _startTime The time at which the payment begins paying out
+  /// @param _endTime The time at which the payment ends paying out
+  /// @param _interval The period of time over which _amounts are paid out
+  /// @param _recipient The recipient of the streaming payment
+  /// @param _tokens The tokens to be paid out
+  /// @param _amounts The amounts to be paid out (per _interval of time)
+  function create(
+    uint256 _fundingPermissionDomainId,
+    uint256 _fundingChildSkillIndex,
+    uint256 _adminPermissionDomainId,
+    uint256 _adminChildSkillIndex,
+    uint256 _domainId,
+    uint256 _startTime,
+    uint256 _endTime,
+    uint256 _interval,
+    address payable _recipient,
+    address[] memory _tokens,
+    uint256[] memory _amounts
+  )
+    public
+    notDeprecated
+    validateFundingPermission(_fundingPermissionDomainId, _fundingChildSkillIndex, _domainId)
+    validateAdministrationPermission(_adminPermissionDomainId, _adminChildSkillIndex, _domainId)
+  {
+    uint256 startTime = (_startTime == 0) ? block.timestamp : _startTime;
+
+    require(_tokens.length == _amounts.length, "streaming-payments-bad-input");
+    require(startTime <= _endTime, "streaming-payments-bad-end-time");
+    require(_interval > 0, "streaming-payments-bad-interval");
+
+    numStreamingPayments++;
+    streamingPayments[numStreamingPayments] = StreamingPayment(_recipient, _domainId, startTime, _endTime, _interval);
+
+    emit StreamingPaymentCreated(msgSender(), numStreamingPayments);
+
+    for (uint256 i; i < _tokens.length; i++) {
+      paymentTokens[numStreamingPayments][_tokens[i]] = PaymentToken(_amounts[i], 0);
+
+      emit PaymentTokenUpdated(msgSender(), numStreamingPayments, _tokens[i], _amounts[i]);
+    }
+
+  }
+
+  /// @notice Claim a streaming payment
+  /// @param _permissionDomainId The domain in which the extension holds the funding & admin permissions
+  /// @param _childSkillIndex The index linking the permissionDomainId to the domainId the payment is in
+  /// @param _fromChildSkillIndex The linking the domainId to the fromPot domain
+  /// @param _toChildSkillIndex The linking the domainId to the toPot domain
+  /// @param _id The id of the streaming payment
+  /// @param _tokens The tokens to be paid out
+  function claim(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _fromChildSkillIndex,
+    uint256 _toChildSkillIndex,
+    uint256 _id,
+    address[] memory _tokens
+  ) public {
+    StreamingPayment storage streamingPayment = streamingPayments[_id];
+
+    require(streamingPayment.startTime < block.timestamp, "streaming-payments-too-soon-to-claim");
+
+    uint256 domainFundingPotId = colony.getDomain(streamingPayment.domainId).fundingPotId;
```

Comment:
Generally speaking, I'm loathe to store the same data in multiple places. This ends up saving gas if you make six claims over the lifetime of the payment, which doesn't seem like it would be uncommon, but still...

Reply:
I'll let you make the call then.